app/api/authn/views.py

```diff
@@ -0,0 +1,67 @@
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.exceptions import ValidationError
+from django.contrib.auth.models import User
+from django.contrib.auth.hashers import make_password
+from rest_framework.permissions import AllowAny
+from django.middleware.csrf import get_token
+from django.http import JsonResponse
+from rest_framework.permissions import AllowAny
+from rest_framework.views import APIView
+from rest_framework.permissions import IsAuthenticated
+
+
+class PasswordResetView(APIView):
+    """
+    A view to reset the user's password directly without email confirmation.
+    Users provide their username, new password, and confirm password.
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, *args, **kwargs):
+
+        # Extract username, new_password, and confirm_password from the request data
+        username = request.data.get("username")
```

Comment:
username shouldn't come from the request object though - this indicates to me that anyone could change anyone's password still?